ui/src/components/DenylistForm/index.jsx

```diff
@@ -219,11 +259,26 @@ export default class DenylistForm extends Component {
             requiresAuth
             disabled={loading}
             variant="round"
-            onClick={this.handleAddressDelete}
+            onClick={onDialogActionOpen}
             classes={{ root: classes.deleteIcon }}>
             <DeleteIcon />
           </Button>
         )}
+        {dialogOpen && (
+          <DialogAction
+            open={dialogOpen}
+            onSubmit={this.handleAddressDelete}
+            onComplete={onDialogActionComplete}
+            onClose={onDialogActionClose}
+            onError={onDialogActionError}
+            error={dialogError}
+            title="Delete Address?"
+            body={
+              <Typography>This will delete {address.notificationAddress}.</Typography>
```

Comment:
The issue you were having is because `{address}` was an object but React can only render strings.